pkg/services/sqlstore/migrations/user_mig.go

```diff
@@ -140,6 +140,19 @@ func addUserMigrations(mg *Migrator) {
 			SQLite(migSQLITEisServiceAccountNullable).
 			Postgres("ALTER TABLE `user` ALTER COLUMN is_service_account DROP NOT NULL;").
 			Mysql("ALTER TABLE user MODIFY is_service_account BOOLEAN DEFAULT 0;"))
+
+	mg.AddMigration("Add uid column to user", NewAddColumnMigration(userV2, &Column{
+		Name: "uid", Type: DB_NVarchar, Length: 40, Nullable: true,
+	}))
+
+	mg.AddMigration("Update uid column values for users", NewRawSQLMigration("").
+		SQLite("UPDATE user SET uid=printf('u%09d',id) WHERE uid IS NULL;").
+		Postgres("UPDATE `user` SET uid='u' || lpad('' || id::text,9,'0') WHERE uid IS NULL;").
+		Mysql("UPDATE user SET uid=concat('u',lpad(id,9,'0')) WHERE uid IS NULL;"))
```

Comment:
I think instead of running this as a one time migration we should run it on every start up for a couple of minors. Then we would handle a upgrade downgrade scenario better. With a one time migration any users added after a downgrade would not have a uid

Reply:
:+1: with the unique index it will cause the DB to error out anyway on downgrade :cry:

We might need have no unique index for a bit as well